src/OrleansProviders/Streams/Common/PooledCache/FixedSizeObjectPool.cs

```diff
@@ -52,16 +52,25 @@ public override T Allocate()
             T obj;
             lock (locker)
             {
-                // if we've used all we are allowed, signal object it has been purged and should be returned to the pool
-                if (usedObjects.Count >= maxObjectCount)
+                // just send signal to EventhubQueueCaches to ask them do a TimeBasedPurge
```

Comment:
nit: unless this is no longer generally usable, the comment about EventHub doesn't apply here, since it might be used for other purposes (in theory)